IdentityCore/src/oauth2/MSIDTokenResponse.h

```diff
@@ -44,7 +44,10 @@
  */
 @property (nonatomic) NSInteger expiresOn;
 @property (nonatomic, nullable) NSString *accessToken;
+// In AT Pop scenario, we need to pass back the tokenType and kid(req_conf)
 @property (nonatomic, nullable) NSString *tokenType;
+@property (nonatomic, nullable) NSString *requestConf;
```

Comment:
Can you give a bit more details about why we need to return request conf in the response? I'd recommend scoping it to broker token responses since this class gets used also for ordinary (no broker) responses.

Reply:
That's in Rohit's design, I think to make sure we got the AT bound to the right key we generated in MSAL.
So in the request chain: 
MSAL ->(kid, tokenType) -> Broker -> (kid, tokenType) -> Server.
Then in the response chain:
Server ->(AT,req_conf, tokenType) -> Broker -> (AT,req_con, tokenType) MSAL.
MSAL needs to check the req_conf in the response matched to make sure the AT Pop is usable.

About the place to put req_conf, I see both AT, tokenType already put in MSIDTokenResponse, so I think it would be easier to put req_conf in that same class. Put req_conf into MSIDAADV2TokenResponse or MSIDAADTokenResponse is ok for this change but I'm afraid later we will lose the visibility that these all 3 fields need to be tied together.

Reply:
Longer run, I think we can split the token response to child classes: AT/RT/ID token responses, then put each class attributes into its own class. We should do it soon when do RT Pop, cause not only AT carrying special attributes but also RT will carry soon. So we need a better way to understand what attribute belongs to what token type.

Reply:
Ok, as long as the parameter is optional. Agree on using AADTokenresponse if it only belongs to AAD.
I also like your long term proposal.